internal/streamdigest/stream_digest.go

```diff
@@ -0,0 +1,40 @@
+package streamdigest
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/containers/image/v5/internal/tmpdir"
+	"github.com/containers/image/v5/types"
+	digest "github.com/opencontainers/go-digest"
+)
+
+// ComputeBlobInfo streams a blob to a temporary file and populates Digest and Size in inputInfo.
+// The temporary file is returned with the file position set to the beginning of the file.
+// It is the caller's responsibility to remove the file when no longer needed.
+// If an error occurs, inputInfo is not modified.
+func ComputeBlobInfo(sys *types.SystemContext, stream io.Reader, inputInfo *types.BlobInfo) (*os.File, error) {
```

Comment:
Non-blocking, this is an internal interface we can change any time if this became important:

A `func ComputeBlobInfo(…) (stream io.Reader, cleanup func(), error)` (with `cleanup()` doing  `streamCopy.Close()` and `os.Remove(streamCopy.Name())`) would encapsulate the implementation, and more importantly simplify callers, further. (The `cleanup` closure could even simplify error handling inside this function.)

Reply:
Changed the interface to return a cleanup closure.